### PR TITLE
Enable support for different style languages through lang attribute

### LINKED ISSUE #17: Support for lang="scss"
I wasted an hour until I realized that this plugin works fine with `<style type="text/scss">`.

I was using `<style lang="scss">`, because that's what the [VueJS plugin](https://github.com/JetBrains/intellij-plugins/tree/master/vuejs) uses, and that's what I used on Vue apps. Also the [Svelte preprocess package](https://www.npmjs.com/package/svelte-preprocess) is documenting `lang="scss"`, so I thought that was the right way to do it.

So, it would be great to add support for `lang="scss"`, since Vue developers are already used to that.

Thanks for your great plugin!

### LINKED ISSUE #29: improvement: highlighting for sass, less, stylus
it would be nice improve highlighting for sass, less, stylus

image

### PR DESCRIPTION
**Pull request by AleksandrSl**
Tested manually with stylus and scss. I think that some automatic tests must be added, will take a look at them in nearest future

Resolves #17
Resolves #29

### CODE REVIEW

File: src/main/java/dev/blachut/svelte/lang/parsing/html/SvelteHtmlLexer.kt [hunk 1]
```diff
@@ -1,15 +1,42 @@
 package dev.blachut.svelte.lang.parsing.html
 
 import com.intellij.lang.HtmlScriptContentProvider
+import com.intellij.lang.Language
 import com.intellij.lang.javascript.JSStubElementTypes
 import com.intellij.lang.javascript.dialects.JSLanguageLevel
 import com.intellij.lang.javascript.types.JSEmbeddedContentElementType
-import com.intellij.lexer.*
+import com.intellij.lexer.HtmlHighlightingLexer
```
Comment (AleksandrSl) [inline src/main/java/dev/blachut/svelte/lang/parsing/html/SvelteHtmlLexer.kt]: Sorry for this, automatic refactor. Can undo this

**Comment by AleksandrSl** [comment]
Related to #17 and I think to #29, it seems to me that IDEA automatically highlight style blocks

**Comment by AleksandrSl** [comment]
Found no tests for this in Vue plugin (
I also think that different template syntaxes like pug and jade can be supported using similar technique

File: src/main/java/dev/blachut/svelte/lang/parsing/html/SvelteHtmlLexer.kt (hunk 1 above)
Comment (tomblachut) [reply src/main/java/dev/blachut/svelte/lang/parsing/html/SvelteHtmlLexer.kt]: Don't worry, it's fine

**Comment by tomblachut** [comment]
Thx again for the contribution. I'll double check highlighting later.

The problem with Jade is that HTML is extended in Svelte, in contrast to style languages that are used as is. How would you implement if blocks and attribute expressions in Pug? This would also increase maintenance cost in the future, IDEA API changes from time to time :/

**Comment by AleksandrSl** [comment]
Yeah, I remembered about custom svelte tags in template on the way home(

**Comment by tomblachut** [comment]
Whoa time flies so fast 😅 

I've checked the PR. Highlighting is off

**Comment by tomblachut** [comment]
Closed by mistake

**Comment by AleksandrSl** [comment]
I will try to fix highlighting in a couple of days!

**Comment by tomblachut** [comment]
Thx 😄

**Comment by AleksandrSl** [comment]
Added the same helper to `BaseSvelteHtmlHighlightingLexer` seems like highlighting works now
image

**Comment by tomblachut** [comment]
Idk why you've reverted accessors. Without them values are set once, upon initialisation and lexers don't work properly.

I've created a patch locally so I'll merge this and add those `get()`s myself.

Thx for contribution!

**Comment by AleksandrSl** [comment]
@tomblachut Oh, I understood the problem, for some reason I thought that 
`get()` and `=` will be equal in this case, now I see that they are not)

**Pull request merged**